### ISSUE TITLE
Streaming continues after client quits

### ISSUE THREAD

**Issue by dob** [livepeer/livepeer-swarm#4]
<a href="https://github.com/ericxtang"><img src="https://avatars3.githubusercontent.com/u/345743?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ericxtang](https://github.com/ericxtang)**
_Tuesday Feb 07, 2017 at 21:06 GMT_
_Originally opened as https://github.com/livepeer/go-ethereum/issues/17_

----

When streaming with OBS, the subscribing server continues to stream packets into the DstVideoChan even after the player quits.  This causes the subsequent view to error out (because for now, every node can only support 1 viewer).

**Comment by dob** [comment]
<a href="https://github.com/dob"><img src="https://avatars3.githubusercontent.com/u/10389?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dob](https://github.com/dob)**
_Friday Feb 24, 2017 at 22:14 GMT_

----

To reproduce, start streaming via OBS, and playing the stream from another node via `livepeer stream`. Kill the ffplayer, and you'll see in the consuming node that it's still receiving packets.

To prevent this, I actually think we need another API call into the node (unsubscribe), which then passes an unsubscribe message to the peer from which it's consuming the stream from. That node should then remove you from it's downstream peers, and then if there are no more downstream peers (and it's not the original broadcaster or consuming the stream itself) it should pass an unsubscribe message back upstream to its peer. 

So to summarize:

1. New API call into the node: `unsubscribeFromStream`
2. New message type: `unsubscribe`
3. Logic for removing downstream peers who unsubscribe, and then deciding whether to unsubscribe yourself if you're a relay node.

There's also the question of how the client should trigger this unsubscribe command. In the case of `livepeer stream`, we can see when ffplay quits, and then call it ourself. In the case of some other client consuming via rtmp or http interface, we don't have this control. So the above idea may not even work.

Maybe if a node is subscribed to a stream, it should set a timeout on consuming packets off of the DstVideoChan. If one hasn't been consumed in `x` seconds, then just unsubscribe from the stream automatically. What do you think Eric?

**Comment by ericxtang** [comment]
Solved by adding unsubscribe in protocol.go
